### PR TITLE
WIP: interrupt gathering PoC

### PR DESCRIPTION
**Pull request by Sergey1011010**
This PR implements a new data enhancement to...

## Categories

- [ ] Bugfix
- [X] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

- `path/to/sample_data.json`

## Documentation

- `path/to/documentation.md`

## Unit Tests

- `path/to/file_test.go`

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://bugzilla.redhat.com/show_bug.cgi?id=???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/controller/periodic/periodic.go [hunk 1]
```diff
@@ -115,32 +115,45 @@ func (c *Controller) Gather() {
...
+	go func() {
+		// once the configChangedChannel is closed by configChangedCloser
+		// (which happens when the Gather function finishes) this goroutine will stop
+		for range configChangedChannel {
+			// TODO: check that our field has changed
+			fmt.Println()
+			fmt.Println("!!! CONFIG WAS CHANGED DURING GATHERING !!!")
+			fmt.Println("cancelling the gathering...")
+			fmt.Println()
+			cancel()
```
Comment (tremes) [inline pkg/controller/periodic/periodic.go:133]: So this is sufficient to stop the gathering in the progress right? :)

Comment (Sergey1011010) [reply pkg/controller/periodic/periodic.go:133]: We also need to stop the workers and so on to stop it even faster, but essentially, yes

**Pull request closed without merging**